File: Dagger/app/src/main/java/com/snappymob/kotlincomponents/di/MainActivityModule.java
package com.snappymob.kotlincomponents.di;

import com.snappymob.kotlincomponents.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class MainActivityModule {

    @ContributesAndroidInjector
    abstract MainActivity contributeMainActivity();

}

File: Dagger/app/src/main/java/com/snappymob/kotlincomponents/di/MainActivityModule.kt
package com.snappymob.kotlincomponents.di

import com.snappymob.kotlincomponents.MainActivity

import dagger.Module
import dagger.android.ContributesAndroidInjector

/**
 * https://github.com/googlesamples/android-architecture-components/tree/master/GithubBrowserSample/app/src/main/java/com/android/example/github/di
 */
@Module
abstract class MainActivityModule {

    @ContributesAndroidInjector
    internal abstract fun contributeMainActivity(): MainActivity

}